internal/cmd/workspaces.go

```diff
@@ -85,11 +86,18 @@ func lsWorkspacesCommand() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			workspaces, err := getWorkspaces(ctx, client, user)
-			if err != nil {
-				return err
+			var workspaces []coder.Workspace
+			if !all {
+				var err error
+				workspaces, err = getWorkspaces(ctx, client, user)
+				if err != nil {
+					return err
+				}
+			} else {
+				// If the user gave the all flag, then filtering by user doesn't make sense.
+				user = ""
 			}
-			if provider != "" {
+			if provider != "" || all {
```

Comment:
This means the `--all` flag only works with the `--provider` flag. I think that's OK for now, but it needs to be made obvious in the flag description (e.g. "requires --provider flag") and there should be a check at the top of the command to ensure `--provider` is set if `--all` is set.

```
$ ./coder envs ls --all
fatal: resource not found
```

Reply:
Doesn't it mean it works _regardless_ of the provider flag? If I call coder with just `--all`, provider is "", and the logic becomes `if "" != "" || true` -> `if false || true` -> `if true`. This was my intent - as an admin, I want to see _all_ workspaces.

But looking at the actual logic/implementation for getWorkspacesByProvider, I think your point still stands. I can fix that.